File: code_samples/ruby/antigravity_automation_api_examples.rb
# Antigravity Automation API — Ruby Code Samples
# Requires: gem install json (stdlib), optionally: gem install faye-websocket eventmachine

require "net/http"
require "json"
require "uri"

BASE_URL = "http://localhost:5000"

def post(endpoint, body = nil)
  uri = URI("#{BASE_URL}#{endpoint}")
  req = Net::HTTP::Post.new(uri, "Content-Type" => "application/json")
  req.body = body.to_json if body
  res = Net::HTTP.start(uri.hostname, uri.port) { |http| http.request(req) }
  data = JSON.parse(res.body)
  puts "#{endpoint} → #{data}"
  data
end

def get(endpoint)
  uri = URI("#{BASE_URL}#{endpoint}")
  res = Net::HTTP.get_response(uri)
  data = JSON.parse(res.body)
  puts "#{endpoint} → #{data}"
  data
end

# ── Automation ──────────────────────────────────────
post("/toggle_auto_run")
post("/toggle_auto_allow")

# ── Remote Control ──────────────────────────────────
post("/send_command",  { text: "Refactor the auth module" })
post("/start-new-chat")
post("/switch_chat",   { title: "Refactoring Auth Module" })

# ── Content ─────────────────────────────────────────
get("/get_command")
post("/update", { title: "Debug Session", content: "User: fix the bug\nAssistant: ..." })

# ── Stats ───────────────────────────────────────────
get("/stats")
post("/track_action", { action: "auto_run" })

# ── WebSocket Stream ────────────────────────────────
# require "faye/websocket"
# require "eventmachine"
#
# EM.run do
#   ws = Faye::WebSocket::Client.new("ws://localhost:9812")
#   ws.on(:message) { |e| puts JSON.parse(e.data) }
#   ws.on(:close)   { EM.stop }
# end

File: code_samples/php/antigravity_automation_api_examples.php
<?php
/**
 * Antigravity Automation API — PHP Code Samples
 * Requires: PHP 7.4+ with cURL extension (built-in)
 */

$BASE_URL = "http://localhost:5000";

function api_post(string $endpoint, ?array $body = null): array {
    global $BASE_URL;
    $ch = curl_init("$BASE_URL$endpoint");
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_POST, true);
    if ($body) {
        curl_setopt($ch, CURLOPT_HTTPHEADER, ["Content-Type: application/json"]);
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
    }
    $response = curl_exec($ch);
    curl_close($ch);
    $data = json_decode($response, true);
    echo "$endpoint → " . json_encode($data) . "\n";
    return $data;
}

function api_get(string $endpoint): array {
    global $BASE_URL;
    $response = file_get_contents("$BASE_URL$endpoint");
    $data = json_decode($response, true);
    echo "$endpoint → " . json_encode($data) . "\n";
    return $data;
}

// ── Automation ──────────────────────────────────────
api_post("/toggle_auto_run");
api_post("/toggle_auto_allow");

// ── Remote Control ──────────────────────────────────
api_post("/send_command",  ["text" => "Refactor the auth module"]);
api_post("/start-new-chat");
api_post("/switch_chat",   ["title" => "Refactoring Auth Module"]);

// ── Content ─────────────────────────────────────────
api_get("/get_command");
api_post("/update", [
    "title"   => "Debug Session",
    "content" => "User: fix the bug\nAssistant: ..."
]);

// ── Stats ───────────────────────────────────────────
api_get("/stats");
api_post("/track_action", ["action" => "auto_run"]);

// ── WebSocket Stream ────────────────────────────────
// For PHP WebSocket, use: composer require textalk/websocket
//
// use WebSocket\Client;
// $client = new Client("ws://localhost:9812");
// while (true) {
//     $msg = $client->receive();
//     $data = json_decode($msg, true);
//     echo "[{$data['title']}] {$data['content']}\n";
// }
